cips/cip-draft_ica.md

```diff
@@ -0,0 +1,917 @@
+
+
+### Integration
+
+The interchain accounts module should be registered as an `AppModule` in the same way all SDK modules are registered on a chain, as well as an `IBCModule`. 
+
+```go
+// app.go
+
+// Register the AppModule for the Interchain Accounts module
+ModuleBasics = module.NewBasicManager(
+  ...
+  ica.AppModuleBasic{},
+  ...
+)
+
+... 
+
+// Add module account permissions for the Interchain Accounts module
+// Only necessary for host chain functionality
+// Each Interchain Account created on the host chain is derived from the module account created
+maccPerms = map[string][]string{
+  ...
+  icatypes.ModuleName:            nil,
+}
+
+...
+
+// Add Interchain Accounts Keepers for each submodule used and the authentication module
+// If a submodule is being statically disabled, the associated Keeper does not need to be added. 
+type App struct {
+  ...
+  ICAHostKeeper       icahostkeeper.Keeper
+  ...
+}
+
+...
+
+// Create store keys for each submodule Keeper and the authentication module
+keys := sdk.NewKVStoreKeys(
+  ...
+  icahosttypes.StoreKey,
+  ...
+)
+
+... 
+
+// Create the scoped keepers for the host submodule
+scopedICAHostKeeper := app.CapabilityKeeper.ScopeToModule(icahosttypes.SubModuleName)
+
+...
+
+// Create the Keeper for the host submodule
+app.ICAHostKeeper = icahostkeeper.NewKeeper(
+  appCodec, keys[icahosttypes.StoreKey], app.GetSubspace(icahosttypes.SubModuleName),
+  app.IBCKeeper.ChannelKeeper, // may be replaced with middleware such as ics29 fee
+  app.IBCKeeper.ChannelKeeper, &app.IBCKeeper.PortKeeper,
+  app.AccountKeeper, scopedICAHostKeeper, app.MsgServiceRouter(),
+)
+
+// Create Interchain Accounts AppModule
+// Since only the host module is registered, nil is passed as the controller keeper
+icaModule := ica.NewAppModule(nil, &app.ICAHostKeeper)
+
+// Create a host IBC module 
+icaHostIBCModule := icahost.NewIBCModule(app.ICAHostKeeper)
+
+// Register host route
+ibcRouter.
+  AddRoute(icahosttypes.SubModuleName, icaHostIBCModule)
+...
+
+// Register Interchain Accounts AppModule
+app.moduleManager = module.NewManager(
+  ...
+  icaModule,
+)
+
+...
+
+// Add Interchain Accounts to begin blocker logic
+app.moduleManager.SetOrderBeginBlockers(
+  ...
+  icatypes.ModuleName,
+  ...
+)
+
+// Add Interchain Accounts to end blocker logic
+app.moduleManager.SetOrderEndBlockers(
+  ...
+  icatypes.ModuleName,
+  ...
+)
+
+// Add Interchain Accounts module InitGenesis logic
+app.moduleManager.SetOrderInitGenesis(
+  ...
+  icatypes.ModuleName,
+  ...
+)
+
+// initParamsKeeper init params keeper and its subspaces
+func initParamsKeeper(appCodec codec.BinaryCodec, legacyAmino *codec.LegacyAmino, key, tkey sdk.StoreKey) paramskeeper.Keeper {
+  ...
+  paramsKeeper.Subspace(icahosttypes.SubModuleName)
+  ...
+}
+```
+
+## Rationale
+
+[COPY FROM GOOGLE DOC ONCE COMPLETE]
+
+## Backwards Compatibility
+
+No backward compatibility issues found
+
+## Test Cases
+
+TBA
```

Comment:
We should get feedback from the Celestia team on what they'd like to see when we present this